flows/engine/engine.go

```diff
@@ -6,12 +6,17 @@ import (
 
 	"github.com/nyaruka/goflow/flows"
 	"github.com/nyaruka/goflow/flows/events"
+	"github.com/nyaruka/goflow/flows/runs"
 )
 
 // StartFlow starts the flow for the passed in contact, returning the created FlowRun
-func StartFlow(env flows.FlowEnvironment, flow flows.Flow, contact *flows.Contact, parent flows.FlowRun, callerEvents []flows.Event, extra json.RawMessage) (flows.Session, error) {
-	// build our run
-	run := flow.CreateRun(env, contact, parent)
+func StartFlow(env flows.SessionEnvironment, flow flows.Flow, contact *flows.Contact, parent flows.FlowRun, callerEvents []flows.Event, extra json.RawMessage, session flows.Session) (flows.Session, error) {
+	if session == nil {
```

Comment:
Hrmm, ya, do agree with that, though this method has a lot of nilable things as well.

Is there ever a time when the both session and parent will be nil at the same time? Or where the session of the parent is not the same as the passed in session?

IE, do we really need session given that we have parent?